doc/source/development/release_process.rst

```diff
@@ -0,0 +1,217 @@
+
+Any committer can perform the following steps to create and call a vote on a proposed release.
+
+Check that no open jira tickets are urgent and currently being worked on.
+Also check with a PMC that there's security vulnerabilities currently being worked on in private.
+
+Perform the Release
+-------------------
+
+Run the following commands to generate and upload release artifacts, to a nexus staging repository and distribution location::
+
+    git clone https://github.com/apache/cassandra-builds.git
+    cd cassandra-build/cassandra-release
+    # edit the variables at the top of `prepare_release.sh`
+    ./prepare_release.sh
+
+If successful, take note of the email text output which can be used in the next section "Call for a Vote".
+
+The ``prepare_release.sh`` script does not yet generate and upload the rpm and deb distribution packages.
+To generate and upload them do::
+
+    cd ../..
+    cassandra-build/build-scripts/cassandra-deb-packaging.sh http://git-wip-us.apache.org/repos/asf?p=cassandra.git <version>-tentative <version>
+    cassandra-build/build-scripts/cassandra-rpm-packaging.sh http://git-wip-us.apache.org/repos/asf?p=cassandra.git <version>-tentative <version>
+
+    svn co https://dist.apache.org/repos/dist/dev/cassandra cassandra-dist-dev
+    mkdir cassandra-dist-dev/<version>
+    cp cassandra-build/docker/*.rpm cassandra-dist-dev/<version>/
+    cp cassandra-build/docker/*.deb cassandra-dist-dev/<version>/
+    svn add cassandra-dist-dev/<version>
+    svn ci cassandra-dist-dev/<version>
+
+
+Call for a Vote
+===============
+
+Fill out the following email template and send to the dev mailing list::
+
+    I propose the following artifacts for release as <version>.
+
+    sha1: <git-sha>
+
+    Git: http://git-wip-us.apache.org/repos/asf?p=cassandra.git;a=shortlog;h=refs/tags/<version>-tentative
+
+    Artifacts: https://repository.apache.org/content/repositories/orgapachecassandra-<nexus-id>/org/apache/cassandra/apache-cassandra/<version>/
+
+    Staging repository: https://repository.apache.org/content/repositories/orgapachecassandra-<nexus-id>/
+
+    The distribution packages are available here: https://dist.apache.org/repos/dist/dev/cassandra/${version}/
+
+    The vote will be open for 72 hours (longer if needed).
+
+    [1]: (CHANGES.txt) https://git1-us-west.apache.org/repos/asf?p=cassandra.git;a=blob_plain;f=CHANGES.txt;hb=<version>-tentative
+    [2]: (NEWS.txt) https://git1-us-west.apache.org/repos/asf?p=cassandra.git;a=blob_plain;f=NEWS.txt;hb=<version>-tentative
+
+
+
+Post-vote operations
+====================
+
+Any PMC can perform the following steps to formalize and publish a successfully voted release.
+
+Publish Artifacts
+-----------------
+
+Run the following commands to publish the voted release artifacts::
+
+    git clone https://github.com/apache/cassandra-builds.git
+    # edit the variables at the top of `finish_release.sh`
+    cassandra-build/cassandra-release/finish_release.sh
+
```

Comment:
What I'm missing from the script is updating the rpm repo. Possibly also signing the packages and repo metadata, see https://github.com/apache/cassandra-builds#signing-packages

Reply:
i'm not sure where/when the correct place to put the `rpmbuild …` command is.

Reply:
At the moment, I copy the rpms from the ```dist/``` directory to the appropriate redhat/NNx location in svn.Then:
```
createrepo .
gpg --detach-sign --armor repodata/repomd.xml
for f in `find repodata/ -name *.bz2`; do gpg --detach-sign --armor $f; done  # thanks to an ignorant script that autogenerates unsigned file warnings on the compressed metadata that is already signed..
```

Reply:
@mshuler i took a stab at updating this, but a few pieces in the puzzle are still unclear to me (like what repodata is? and steps in 'Publish Artifacts' vs 'Sign and Upload Distribution Packages to Bintray'?)